### ISSUE TITLE
Handle [UInt8] on whiteboard slot

### ISSUE THREAD

**Issue by LewisKrygger** [mipalgu/swift-whiteboard#1]
Currently posting an array of UInt8 to a whiteboard slot will eventually cause a bad access crash, this behaviour is expected, but having this functionality would be very useful in certain situations. 

@rhx in our meeting today you have suggested this could be implemented. @Morgan2010  has also done a lot of this work already on the CFE and suggested discussion with him.

**Comment by rhx** [comment]
Can you pull in the development branch and check that the `post(array:)` and `getArray()` methods work for you?

**Comment by rhx** [comment]
There is also now a low-level `post(elements:)` for fixed-size buffers.  Be careful, though, this does not store the element count anywhere on the whiteboard, so its counterpart `getElements()` doesn't actually do any copying, requiring you to do the copying yourself from the returned `UnsafeBufferPointer` (which points directly into the whiteboard memory for the returned message).

**Comment by LewisKrygger** [comment]
I can confirm my tests are now working with [UInt8] data.

**Comment by CarlLusty** [comment]
> Be careful, though, this does not store the element count anywhere on the whiteboard

Why not store the length in the last byte of the whiteboard message slot? Then you could return a copy instead of the whiteboard memory.

**Comment by Morgan2010** [comment]
> > Be careful, though, this does not store the element count anywhere on the whiteboard
> 
> Why not store the length in the last byte of the whiteboard message slot? Then you could return a copy instead of the whiteboard memory.

In conjunction with what Carl is saying, when you use the latest stuff (not released) from our abstraction of the cFS, the byte count is stored in the first few bytes of the array. That way you can convert to your array from other pointers that are not Buffer pointers (eg. raw pointers). This will be needed if you send your data through the cFS Pipes. I would imagine that this is an option for the whiteboard as well.

**Comment by rhx** [comment]
Yes, the (variable-length) array versions store the length in the first two bytes (UInt16).  Only the raw buffer-pointer ones allow you to manage this on your own (e.g. if you know how many elements you have and won't need to store this).
